### ISSUE TITLE
Custom Admin URL path hardcoded in build

### ISSUE THREAD

**Issue by ssshammi** [strapi/strapi#12746]
After yarn build the files when transferred to production without using env variables. admin is still calling /admin/project-type and /admin/init
 image screenshoot.
[https://d38xoolefqxt5u.cloudfront.net/original/2X/e/ed281bacfc678a12d5334136693093bee29e9743.png](url)

This is because the path is hardcoded in node_modules\@strapi\admin\admin\src\index.js 
```

const run = async () => {
  try {
    const {
      data: {
        data: { isEE, features },
      },
    } = await axiosInstance.get('/admin/project-type');
```

### Expected behavior
Once build is ready and transferred to a subdomain or a folder the hardcoded path admin should go.

- Node.js version: 16

- Strapi version:4.1.2
- Database:Mysql
- Operating system: Windows

**Comment by derrickmehaffy** [comment]
Duplicate of:

https://github.com/strapi/strapi/issues/10867
https://github.com/strapi/strapi/issues/9542
https://github.com/strapi/strapi/issues/9027
https://github.com/strapi/strapi/issues/7419
https://github.com/strapi/strapi/issues/2122
https://github.com/strapi/strapi/issues/1452

---

The fact the admin url is hard coded at build time is 100% intended in order to properly support serving the admin **outside of Strapi's backend**. The Admin panel is react and thus CSR so it needs to know how to contact the backend. To ease complexity in the code and loading of env vars during build vs run time it is hard coded based on the value of the `./config/server.js` => url key and/or the `./config/admin.js` => url key (plus any `./config/env/*/*.js` environment settings).

We have no plans to change this at the moment.

**Comment by wolodek** [comment]
Actually as PUBLIC_URL is statically compiled into the build you can override it before serving the app with simple replace script. We are having one build step in CI/CD and serve same package across many environemnts. I created a gist which can be run in some after-deploy hook to resolve this issue
https://gist.github.com/wolodek/973309c8540437e0cf33ca8b8148023e
